api/src/main/java/org/motechproject/nms/api/web/UserController.java

```diff
@@ -137,7 +135,7 @@ public UserResponse getUserDetails(@PathVariable String serviceName,
 
         // If the user does not have a language location code we want to return the allowed language location
         // codes for the provided circle, or all if no circle was provided
-        List<Language> languages = new ArrayList<>();
+        Set<Language> languages = new HashSet<>();
         if (user.getLanguageLocationCode() == null && circleObj != null) {
```

Comment:
Yeah.  I was thinking of adding an assert above.

First we verify we get one of the three possible services, if not we bail.
Then we call either getFLWUser or getKilkariUser, both of which will either
throw or return a user.  So it isn't actually possible.

On Thu, Jul 23, 2015 at 1:41 PM, kosh <notifications@github.com> wrote:

> In api/src/main/java/org/motechproject/nms/api/web/UserController.java
> <https://github.com/motech-implementations/mim/pull/525#discussion_r35368750>
> :
>
> > @@ -137,7 +135,7 @@ public UserResponse getUserDetails(@PathVariable String serviceName,
> >
> >          // If the user does not have a language location code we want to return the allowed language location
> >          // codes for the provided circle, or all if no circle was provided
> > -        List<Language> languages = new ArrayList<>();
> > +        Set<Language> languages = new HashSet<>();
> >          if (user.getLanguageLocationCode() == null && circleObj != null) {
>
> Unrelated to this, but intellij warns of a NPE here since there is a
> possibility that the user object was never set.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/motech-implementations/mim/pull/525/files#r35368750>.
>